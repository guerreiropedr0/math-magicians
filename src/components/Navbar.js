import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="navbar">
    <h1>Math Magicians</h1>
    <ul className="nav-links">
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="/calculator">Calculator</Link>
      </li>
      <li>
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);
export default Navbar;
