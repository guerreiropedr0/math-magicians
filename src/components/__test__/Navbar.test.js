import { screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import Navbar from '../Navbar';
import '@testing-library/jest-dom';

const MockNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

describe('Navbar component', () => {
  it('Navbar element is in the document', () => {
    render(<MockNavbar />);
    const HEADING_ELEMENT = screen.getByRole('heading');
    expect(HEADING_ELEMENT).toMatchSnapshot();
  });
});
