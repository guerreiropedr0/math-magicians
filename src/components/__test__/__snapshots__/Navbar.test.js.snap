// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Navbar element is in the document 1`] = `
<h1>
  Math Magicians
</h1>
`;

exports[`Navbar component should click a link and change the page 1`] = `
<div
  class="calculator"
  data-testid="calc"
>
  <p
    class="output"
  >
    0
  </p>
  <button
    type="button"
  >
    AC
  </button>
  <button
    type="button"
  >
    +/-
  </button>
  <button
    type="button"
  >
    %
  </button>
  <button
    id="operators"
    type="button"
  >
    รท
  </button>
  <button
    type="button"
  >
    7
  </button>
  <button
    type="button"
  >
    8
  </button>
  <button
    type="button"
  >
    9
  </button>
  <button
    id="operators"
    type="button"
  >
    x
  </button>
  <button
    type="button"
  >
    4
  </button>
  <button
    type="button"
  >
    5
  </button>
  <button
    type="button"
  >
    6
  </button>
  <button
    id="operators"
    type="button"
  >
    -
  </button>
  <button
    type="button"
  >
    1
  </button>
  <button
    type="button"
  >
    2
  </button>
  <button
    type="button"
  >
    3
  </button>
  <button
    id="operators"
    type="button"
  >
    +
  </button>
  <button
    id="zero"
    type="button"
  >
    0
  </button>
  <button
    type="button"
  >
    .
  </button>
  <button
    id="operators"
    type="button"
  >
    =
  </button>
</div>
`;
